### ISSUE TITLE
Unsupported URL Type

### ISSUE THREAD

**Issue by ZillDing** [npm/npm#11375]
Met this error when installing some private module using npm v3 through
`npm install -S git+ssh://user@domain.com:/path/to/module.git`

The dependency added by the above command will cause following npm error

```
npm ERR! Windows_NT 6.1.7601
npm ERR! node v5.5.0
npm ERR! npm  v3.6.0

npm ERR! Unsupported URL Type: ssh://git@github.com:/repo/webapps/private-module.git
npm ERR!
npm ERR! If you need help, you may report this error at:
npm ERR!     <https://github.com/npm/npm/issues>

npm ERR! Please include the following file with any support request:
...
```

This looks like a similar issue to [https://github.com/npm/npm/issues/6422](https://github.com/npm/npm/issues/6422) which was resolved for npm v2

**Comment by KenanY** [comment]
@ZillDing Is this a duplicate of https://github.com/npm/npm/issues/9679?

**Comment by ZillDing** [comment]
whoops~ indeed~ please refer #9679

**Comment by ZillDing** [comment]
It seems the original issue: #9679 is closed. However, I met this problem today with the latest stable build.

And there is a little difference. The `npm install --save git+ssh://...` has no problem. It's the link `ssh://...` saved into package.json file that causes all following npm commands the error I posted previously.

**Comment by KenanY** [comment]
@ZillDing `npm --version`?

**Comment by ZillDing** [comment]
```
npm ERR! Windows_NT 6.1.7601
npm ERR! node v5.5.0
npm ERR! npm  v3.6.0
```

**Comment by KenanY** [comment]
@ZillDing Bug was fixed in `3.7.0`.

**Comment by KenanY** [comment]
(you should skip `3.7.0` in favor of `3.7.1` though because a nasty bug was fixed)

**Comment by ZillDing** [comment]
@KenanY Thanks for the fix. I will try as soon as  `3.7.0` is released.

**Comment by aaronup** [comment]
We are still getting this error with npm 3.7.1
```
npm ERR! node v4.3.0
npm ERR! npm  v3.7.1

npm ERR! Unsupported URL Type: ssh+git://PROJECT_REPO.git
npm ERR!
npm ERR! If you need help, you may report this error at:
npm ERR!     <https://github.com/npm/npm/issues>
```

**Comment by othiym23** [comment]
Closing as resolved (`npm@latest` is now `npm@3.7.2`).
